tfac/figures/figure7.py

```diff
@@ -0,0 +1,320 @@
+"""
+Creates Figure 7 -- Imputation Effectiveness
+"""
+import matplotlib
+from matplotlib.patches import Patch
+import numpy as np
+from os.path import abspath, dirname
+import pandas as pd
+from scipy.stats import pearsonr
+from tensorpack import perform_CMTF
+from tensorly import cp_to_tensor
+
+from .common import getSetup
+from ..dataImport import form_tensor, import_cytokines
+
+COLOR_CYCLE = matplotlib.rcParams['axes.prop_cycle'].by_key()['color'][2:]
+PATH_HERE = dirname(dirname(abspath(__file__)))
+
+
+def impute_cv():
+    """
+    Imputes cytokines for each subject via cross-validation.
+
+    Parameters:
+        None
+
+    Returns:
+        tensor (numpy.array): cytokine measurements
+        imputed (numpy.array): cytokine measurements predicted via CMTF
+    """
+    tensor, matrix, patient_data = form_tensor()
+    both_cyto = patient_data.loc[:, 'type'].str.contains('0Serum1Plasma')
```

Comment:
Should be! This line should restrict only to patients with both serum and plasma data.